src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/MemoryStreamAdapter.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+
+namespace System.Runtime.Serialization
+{
+    // needs to be a separate class so it can get its own namespace
+    [DataContract(Name = nameof(MarshalByRefObject), Namespace = Globals.DataContractXsdBaseNamespace + "System")]
+    internal abstract class MarshalByRefObjectAdapter
+    {
```

Comment:
I don't think you need this. DataContractSerializer drops any data it doesn't recognize and as we're not preserving this in MemoryStream or serializing any value other than null, I think it's safe to not have this base class.

Reply:
Deserialization seems to require this. DCS's behavior seems to be that unrecognized data is only allowed to come at the _end_ of the stream. And since Full FX was putting this value into the serialized payload, we need to ensure it's read before any of the real `MemoryStream` data is read. Maybe I'm missing something here?

Reply:
From the documentation [here](https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/data-contract-versioning#adding-and-removing-data-members):

> When a type with an extra field is deserialized into a type with a missing field, the extra information is ignored.

DCS ignores any unrecognized data. The problem you can get is out of order data. So if you are expecting A followed by B but get B followed by A in your input, it will read B and presume A is missing, then when it gets to A, consider it extra data and drop it (as the data needs to be ordered). Not having the base type is basically a backwards version of backwards compatibility. It's like saying the base type wasn't there on an earlier version and now you added it, DCS won't break with the "old" version deserializing. Did you have a problem trying to remove it?

It's not a big deal to leave it in, just not quite as clean as it could be.

Reply:
That's interesting. It must be a nuance as a result of the Serialization attribute instead of the DataContract attribute. You could fold it up into the same class and use the Order property to make it first (base types get serialized first which puts it at the front so Order property overrides the default ordering), but I don't think there's any performance benefit as there's no difference in number of objects allocated. I'm okay with how things currently are.

Reply:
> You could fold it up into the same class and use the Order property to make it first

Tried that, but there seems to be no way to control the namespace. This field has a different xmlns than all other fields being serialized. Is there a per-field attribute I wasn't seeing?

Reply:
You're right. It's the XmlElementAttribute which allows specifying a different NS. I had a look at the code which writes the XML and the NS is pulled from the class contract and not the data member. So looks like this is the only way to do it.